deltacat/compute/compactor_v2/utils/exception_handler.py

```diff
@@ -0,0 +1,138 @@
+import botocore
+import ray
+from ray.exceptions import (
+    RayError,
+    RuntimeEnvSetupError,
+    WorkerCrashedError,
+    NodeDiedError,
+    OutOfMemoryError,
+)
+from pyarrow.lib import ArrowException, ArrowInvalid, ArrowCapacityError
+from deltacat.exceptions import (
+    DependencyPyarrowError,
+    DependencyRayRuntimeSetupError,
+    DependencyPyarrowInvalidError,
+    DependencyPyarrowCapacityError,
+    DependencyDaftError,
+    DependencyDaftTransientError,
+    GeneralAssertionError,
+    DependencyRayWorkerDiedError,
+    DependencyRayOutOfMemoryError,
+    DependencyRayError,
+)
+from typing import Callable
+from deltacat.utils.ray_utils.runtime import (
+    get_current_ray_task_id,
+)
+from daft.exceptions import DaftTransientError, DaftCoreException
+import tenacity
+
+# List of retryable errors that can be automatically retried by Ray remote tasks
+RAY_TASK_RETRYABLE_ERROR_CODES = [
+    botocore.exceptions.ConnectionError,
+    botocore.exceptions.HTTPClientError,
+    ConnectionError,
+    TimeoutError,
+    DaftTransientError,
+    tenacity.RetryError,
+]
+
+
+def handle_exception(func: Callable):
```

Comment:
```suggestion
def categorize_errors(func: Callable):
```

Handling means doing something about each error. We aren't really handling the exceptions, are we? We are merely categorizing them.